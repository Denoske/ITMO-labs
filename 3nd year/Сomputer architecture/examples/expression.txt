(defv a 1)
(defv b 1)
(defv c 1)
(defv r 1)
(defv sum 0)
(setq sum (+ a b c r))
(print sum)
