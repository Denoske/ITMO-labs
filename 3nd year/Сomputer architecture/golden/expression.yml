source: |-
  (defv a 1)
  (defv b 1)
  (defv c 1)
  (defv r 1)
  (defv sum 0)
  (setq sum (+ a b c r))
  (print sum)
input: |
log: |
  DEBUG    root:machine.py:384 {TICK: 0, PC: 100, ADDR: 100, R0: 0, R1: 0, R2: 0, R3: 0, R4: 0, R5: 0 } mov r0 0
  DEBUG    root:machine.py:390 {TICK: 2, PC: 101, ADDR: 101, R0: 1, R1: 0, R2: 0, R3: 0, R4: 0, R5: 0 } mov r1 1
  DEBUG    root:machine.py:390 {TICK: 4, PC: 102, ADDR: 102, R0: 1, R1: 1, R2: 0, R3: 0, R4: 0, R5: 0 } mov r2 2
  DEBUG    root:machine.py:390 {TICK: 6, PC: 103, ADDR: 103, R0: 1, R1: 1, R2: 1, R3: 0, R4: 0, R5: 0 } mov r3 3
  DEBUG    root:machine.py:390 {TICK: 8, PC: 104, ADDR: 104, R0: 1, R1: 1, R2: 1, R3: 1, R4: 0, R5: 0 } add r0 r1 -> r0
  DEBUG    root:machine.py:390 {TICK: 10, PC: 105, ADDR: 105, R0: 2, R1: 1, R2: 1, R3: 1, R4: 0, R5: 0 } add r0 r2 -> r0
  DEBUG    root:machine.py:390 {TICK: 12, PC: 106, ADDR: 106, R0: 3, R1: 1, R2: 1, R3: 1, R4: 0, R5: 0 } add r0 r3 -> r0
  DEBUG    root:machine.py:390 {TICK: 14, PC: 107, ADDR: 107, R0: 4, R1: 1, R2: 1, R3: 1, R4: 0, R5: 0 } mov 4 r0 -> r0
  DEBUG    root:machine.py:390 {TICK: 16, PC: 108, ADDR: 108, R0: 4, R1: 1, R2: 1, R3: 1, R4: 0, R5: 0 } mov r1 4
  DEBUG    root:machine.py:390 {TICK: 18, PC: 109, ADDR: 109, R0: 4, R1: 4, R2: 1, R3: 1, R4: 0, R5: 0 } print r1 -> r1
  DEBUG    root:machine.py:122 output: '' << '4'
  DEBUG    root:machine.py:390 {TICK: 20, PC: 110, ADDR: 110, R0: 4, R1: 4, R2: 1, R3: 1, R4: 0, R5: 0 } hlt 
  INFO     root:machine.py:395 output_buffer: '4'
output: |
  source LoC: 24 code instr: 11
  ============================================================
  4
  instr_counter:  10
code: |-
  [
      {
          "opcode": "mov",
          "arg1": "r0",
          "arg2": 0,
          "term": "mov"
      },
      {
          "opcode": "mov",
          "arg1": "r1",
          "arg2": 1,
          "term": "mov"
      },
      {
          "opcode": "mov",
          "arg1": "r2",
          "arg2": 2,
          "term": "mov"
      },
      {
          "opcode": "mov",
          "arg1": "r3",
          "arg2": 3,
          "term": "mov"
      },
      {
          "opcode": "add",
          "args": [
              "r0",
              "r1"
          ],
          "res_reg": "r0",
          "term": "add"
      },
      {
          "opcode": "add",
          "args": [
              "r0",
              "r2"
          ],
          "res_reg": "r0",
          "term": "add"
      },
      {
          "opcode": "add",
          "args": [
              "r0",
              "r3"
          ],
          "res_reg": "r0",
          "term": "add"
      },
      {
          "opcode": "mov",
          "arg1": 4,
          "arg2": "r0",
          "res_reg": "r0",
          "term": "mov"
      },
      {
          "opcode": "mov",
          "arg1": "r1",
          "arg2": 4,
          "term": "mov"
      },
      {
          "opcode": "print",
          "arg1": "r1",
          "res_reg": "r1",
          "term": "print"
      },
      {
          "opcode": "hlt",
          "term": "hlt"
      },
      {
          "data": {
              "0": 1,
              "1": 1,
              "2": 1,
              "3": 1,
              "4": 0
          }
      }
  ]